### DISCUSSION TITLE
KZG Commitment 中的一个双线性映射没明白

### DISCUSSION THREAD

**Discussion by Demian101** [zkp-co-learning/zkp-co-learn#98] [Q&A（提问题在隔壁~）] [answered]
几个前置信息： 

-  $\color{red}s \cdot G$  就是 $\color{red}VK$,  我们上面提到过:   `setup` 的 2 个集合 :  第一个集合为 $PK$,  第二个集合是 $\color{red}VK$  ( $VK$ 就是 $PK$  里的某个随机点)
-  $e(G, \ G)$  是公开的
-  $f(i)$  是 `Prover` 发送过来的在 $i$  的打开
-  ${\color{blue}q_i(s) \cdot G}$  是商多项式在 KZG 秘密 $s$ 处的 Commitment

![image](https://github.com/Antalpha-Labs/zkp-co-learn/assets/33189338/19d32aff-181f-416e-90f1-5accb9b91f97)

没明白图中画圈部分的双线性映射是怎么过去的，万望指点一二！

**Comment by Howard-Hu** [comment, marked as answer]
![image](https://github.com/Antalpha-Labs/zkp-co-learn/assets/24404366/2f96998a-3e34-4e8a-a2ba-00d8898fc52d)
如图，根据GB/T 38635.1--2020 《信息安全技术 SM9标识密码算法 第1部分：总则》
6.1 节双线性对的定义可知，双线性对具有双线性性。

由双线性性可知 
 
$$ 
e(\frac{f(s)-f(i)}{s-i}\cdot G,(s-i)\cdot G)=e(G,G)^{\frac{f(s)-f(i)}{s-i} \cdot (s-i)}=e(G,G)^{f(s)-f(i)}
$$

【推荐阅读】
1. [Pairs for beginners by Craig Costello](https://static1.squarespace.com/static/5fdbb09f31d71c1227082339/t/5ff394720493bd28278889c6/1609798774687/PairingsForBeginners.pdf)

**Comment by Demian101** [comment]
感谢笃行！
